cmd/crank/update.go

```diff
@@ -0,0 +1,144 @@
+}
+
+// Run runs the Configuration update cmd.
+func (c *updateConfigCmd) Run(k *kong.Context) error {
+	pkgName := c.Name
+	pkgReference, _ := name.ParseReference(c.Name)
+	newPkg := ""
+	if strings.HasPrefix(c.Tag, "sha256") {
+		newPkg = pkgReference.Context().Digest(c.Tag).Name()
+	} else {
+		newPkg = pkgReference.Context().Tag(c.Tag).Name()
+	}
+	cr := &v1.Configuration{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pkgName,
+		},
+		Spec: v1.ConfigurationSpec{
+			PackageSpec: v1.PackageSpec{
+				Package: newPkg,
+			},
+		},
+	}
+	if c.RevisionHistoryLimit != nil {
+		cr.Spec.PackageSpec.RevisionHistoryLimit = c.RevisionHistoryLimit
+	}
+	if c.ManualActivation != nil {
+		rap := v1.AutomaticActivation
+		if *c.ManualActivation {
+			rap = v1.ManualActivation
+		}
+		cr.Spec.PackageSpec.RevisionActivationPolicy = &rap
+	}
+	if c.PackagePullSecrets != nil {
+		packagePullSecrets := make([]corev1.LocalObjectReference, len(*(c.PackagePullSecrets)))
+		for i, s := range *(c.PackagePullSecrets) {
+			packagePullSecrets[i] = corev1.LocalObjectReference{
+				Name: s,
+			}
+		}
+		cr.Spec.PackageSpec.PackagePullSecrets = packagePullSecrets
+	}
+	kube := typedclient.NewForConfigOrDie(ctrl.GetConfigOrDie())
+	req := new(bytes.Buffer)
+	_ = json.NewEncoder(req).Encode(cr)
+	res, err := kube.Configurations().Patch(context.Background(), cr.Name, types.StrategicMergePatchType, req.Bytes(), metav1.PatchOptions{})
+	if err != nil {
+		return errors.Wrap(warnIfNotFound(err), "cannot update configuration")
+	}
+	_, err = fmt.Fprintf(k.Stdout, "%s/%s updated\n", strings.ToLower(v1.ConfigurationGroupKind), res.GetName())
+	return err
+}
+
+// updateProviderCmd update a Provider.
+type updateProviderCmd struct {
+	Name string `arg:"" help:"Name of Provider."`
+	Tag  string `arg:"" help:"Updated tag for Provider package."`
+
+	RevisionHistoryLimit *int64 `short:"rl" help:"Revision history limit."`
+	ManualActivation     *bool  `short:"m" help:"Enable manual revision activation policy."`
+}
+
+// Run runs the Provider update cmd.
+func (c *updateProviderCmd) Run(k *kong.Context) error {
+	pkgName := c.Name
+	pkgReference, _ := name.ParseReference(c.Name)
+	newPkg := ""
+	if strings.HasPrefix(c.Tag, "sha256") {
+		newPkg = pkgReference.Context().Digest(c.Tag).Name()
+	} else {
+		newPkg = pkgReference.Context().Tag(c.Tag).Name()
+	}
+	cr := &v1.Provider{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: pkgName,
+		},
+		Spec: v1.ProviderSpec{
+			PackageSpec: v1.PackageSpec{
+				Package: newPkg,
+			},
+		},
+	}
+	if c.RevisionHistoryLimit != nil {
+		cr.Spec.PackageSpec.RevisionHistoryLimit = c.RevisionHistoryLimit
+	}
+	if c.ManualActivation != nil {
+		rap := v1.AutomaticActivation
+		if *c.ManualActivation {
+			rap = v1.ManualActivation
+		}
+		cr.Spec.PackageSpec.RevisionActivationPolicy = &rap
+	}
+	kube := typedclient.NewForConfigOrDie(ctrl.GetConfigOrDie())
+	req := new(bytes.Buffer)
+	_ = json.NewEncoder(req).Encode(cr)
```

Comment:
`json.Marshal` will give you `bytes` back so it is likely simpler to use in this case